pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -359,7 +362,11 @@ public ByteBuffer getState(String key) {
             }
         }
 
-        return producer.sendAsync(object).thenApply(msgId -> null);
+        return producer.sendAsync(object).exceptionally(e -> {
```

Comment:
This is returning a future that is masking the eventual error. It should be something like : 

```java
CompletableFuture<Void> future = producer.sendAsync(object).thenApply(msgId -> null);
future.exceptionally(e -> {
     this.statsManager.incrSysExceptions(e);
     logger.error("Failed to publish to topic {} with error {}", topicName, e);
     return null;
});
return future;
```

Reply:
Fixed